### PR TITLE
feat: Add other_Markdown a markdown renderer

### PR DESCRIPTION
**Pull request by jbrundage**
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [x] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

**Comment by jbrundage** [comment]
image

File: packages/other/src/Markdown.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const marked = _marked;
+
+export class Markdown extends HTMLWidget {
+    constructor() {
+        super();
+    }
+
+    update(domNode, element) {
+        super.update(domNode, element);
+        element.html(marked(this.text()));
```
Comment (GordonSmith) [inline packages/other/src/Markdown.ts]: Does vscroll work for this?

File: packages/other/src/Markdown.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    }
+}
+Markdown.prototype._class += " other_Markdown";
+
+Markdown.prototype.publish("text", "", "string", "Sets text to be rendered as markdown text", null, { tags: ["Basic"] });
+
+export interface Markdown {
+    text(): string;
+    text(_: string): this;
```
Comment (GordonSmith) [inline packages/other/src/Markdown.ts]: "text" or "markdown"?  Text is probably ok as markdown is designed to be readable as text also...  (the markdown editor supports both).

Comment (jbrundage) [reply packages/other/src/Markdown.ts]: I went with `text` because it is/was the most intuitive name that came to mind.

Comment (GordonSmith) [reply packages/other/src/Markdown.ts]: I think the base class already has a "text" method - in which case (for consistency) you could make this `markdown`?

**Comment by GordonSmith** [comment]
Travis failing...

**Comment by jbrundage** [comment]
@GordonSmith passing TravisCI now

File: demos/gallery/samples/other/Markdown.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Markdown } from "@hpcc-js/other";
+
+new Markdown()
+    .target("target")
+    .markdown(`# h1 Heading
```
Comment (GordonSmith) [inline demos/gallery/samples/other/Markdown.js]: Minor - but we are preferring:
```
    .markdown(`\
# h1 Heading
```

File: packages/other/src/Markdown.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { HTMLWidget } from "@hpcc-js/common";
+import * as _marked from "marked";
+
+const marked = _marked;
```
Comment (GordonSmith) [inline packages/other/src/Markdown.ts]: Would be simpler to just `import * as marked from "marked"`?

File: tests/test-other/src/other.spec.ts [hunk 5]
```diff
@@ -26,6 +26,8 @@ describe("@hpcc-js/other", () => {
                                     .data(data.HeatMap.simple.data)
                                 );
                                 break;
+                            case Markdown:
+                                break;
```
Comment (GordonSmith) [inline tests/test-other/src/other.spec.ts]: Add a test.

File: packages/other/src/Markdown.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    update(domNode, element) {
+        super.update(domNode, element);
+        element.html(marked(this.markdown()));
+    }
+}
+Markdown.prototype._class += " other_Markdown";
+
+Markdown.prototype.publish("markdown", "", "string", "Sets the string to be rendered as markdown text", null, { tags: ["Basic"] });
+
+export interface Markdown {
```
Comment (GordonSmith) [inline packages/other/src/Markdown.ts]: Move interface above PP for consistency

**Comment by GordonSmith** [comment]
@jbrundage bump

**Comment by GordonSmith** [comment]
@jbrundage bump II

File: demos/gallery/samples/other/Markdown.js (hunk 3 above)
Comment (GordonSmith) [reply demos/gallery/samples/other/Markdown.js]: @jbrundage bump

File: packages/other/src/Markdown.ts (hunk 4 above)
Comment (GordonSmith) [reply packages/other/src/Markdown.ts]: @jbrundage bump

File: tests/test-other/src/other.spec.ts (hunk 5 above)
Comment (GordonSmith) [reply tests/test-other/src/other.spec.ts]: @jbrundage bump

File: packages/other/src/Markdown.ts (hunk 6 above)
Comment (GordonSmith) [reply packages/other/src/Markdown.ts]: @jbrundage bump

**Pull request closed without merging**